docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+- prevents accidental conversion to unexpected types (which may be problematic eg. when using the `===` operator)
+- makes the application design clearer and future changes easier
+
+**Table of Contents:**
+- [Money class](#money-class)
+- [Money in Forms](#money-in-forms)
+- [Money in Twig Templates](#money-in-twig-templates)
+- [Money in Javascript](#money-in-javascript)
+- [Money in Doctrine](#money-in-doctrine)
+- [Unit and Functional Tests](#unit-and-functional-tests)
+- [Price class](#price-class)
+
+## Money class
+
+`Money` is an immutable [value object](https://codete.com/blog/value-objects/).
+
+It uses a decimal representation of the money amount and it does not contain any reference to the used currency.
+You can get the decimal representation as a `string` via the `getAmount` method.
+
+*Note: If in doubt about the results of any method, you can take a look at [its unit tests](/packages/framework/tests/Unit/Component/Money/MoneyTest.php) which contain many examples of the class' behavior.*
+
+### Construction
+
+`Money` can be constructed directly from integers and numeric strings as they are able to represent decimal numbers precisely: `$tenDollars = Money::create(10)`, `$oneAndHalfEuro = Money::create('1.20')`.
+
+It may be also constructed from floating point numbers, but the scale (number of decimal places) must be explicitly specified: `Money::createFromFloat($floatFromExternalSource, 2)`.
+
+Zero amount of money can be constructed by calling `Money::zero()`.
+
+### Computation
+
+To compute with monetary values you have to use the object's methods instead of arithmetic operators (`+`, `-`, `*`, `/`):
+
+- `Money::add(Money $addend) : Money`
+- `Money::subtract(Money $subtrahend) : Money`
+- `Money::multiply(int|string $multiplier) : Money`
+- `Money::divide(int|string $divisor, int $scale) : Money`
+
+For addition and subtraction, the other parameter has to be also a `Money` instance.
+For multiplication and division, the other parameter has to be an integer or a numeric string (as they are able to represent decimal numbers precisely), not a float.
+
+The scale (number of decimal places) of the result is assigned automatically to all operations except division, keeping the results as precise as possible.
+Results of a division may be inexpressible with a finite decimal (eg. 1 / 3 = 0.3333...), so it's up to the user to specify the requested scale.
+- scale of the result of `add` and `subtract` is the *maximal scale* of both money values
+- scale of the result of `multiply` is the *sum of scales* of both money values
+- scale of the result of `divide` must be *explicitly specified*, the last decimal place will be rounded to minimize the error
+
+*Note: The scale of the money amount is always preserved - `getAmount` will use all decimal places of its scale (eg. zero money with scale 6 would return `0.000000`).*
+
+### Rounding
```

Comment:
Please add 4 examples of rounding here.
* 1.494
* 1.055
* and both with 2 different scales - 1 and 2

Reply:
scale is really a tricky concept :-/

Reply:
Do you really think that rounding needs more explanation?

I've added a note with a link to more examples:
```diff
- This behavior is consistent with `PHP_ROUND_HALF_UP` rounding mode.
+ This behavior is consistent with `PHP_ROUND_HALF_UP` rounding mode, which is the default mode for [the `round` function](http://php.net/manual/en/function.round.php).
```

Reply:
This is better, but there is problem with terms. We are using term `scale`, but in php documentation there is used term `precision`, so it is difficult to understand what do you want to express.
I would really appreciate as same examples as in the php documentation http://php.net/manual/en/function.round.php#refsect1-function.round-examples

Reply:
I think it's sufficient. I don't want to include that much examples, if you don't mind.